prime-router/src/main/kotlin/azure/BlobAccess.kt

```diff
@@ -234,6 +242,75 @@ class BlobAccess() : Logging {
             val previousBlobItemVersions: List<BlobItem>?,
         )
 
+        enum class SchemaType {
+            FHIR_FHIR,
+            FHIR_HL7,
+        }
+
+        /**
+         * Downloads the content of each blob in the [directory] with [blobContainerInfo] takes the input file, runs
+         * the transform on it, and compares it to the output validating that they are the same.
+         */
+        fun validateSchemas(
+            directory: String,
+            blobContainerInfo: BlobContainerMetadata = defaultBlobMetadata,
+            schemaType: SchemaType,
+        ): Boolean {
+            val sourceContainer = getBlobContainer(blobContainerInfo)
+            val blobs =
+                listBlobs(directory, blobContainerInfo, false)
+            val transforms = blobs.filter { it.currentBlobItem.name.endsWith(".yml") }
+            transforms.forEach { currentTransform ->
+                val currentTransformName = currentTransform.currentBlobItem.name
+
+                val transformBlobClient = sourceContainer.getBlobClient(currentTransformName)
+                val transform = transformBlobClient.downloadContent()
+
+                val regex = Regex("(.*/).*")
+                val path = regex.find(currentTransformName)!!.groups[1]!!.value
+                val inputBlobClient = sourceContainer.getBlobClient(path + "input.fhir")
+                val input = inputBlobClient.downloadContent()
+                val inputBundle = FhirTranscoder.decode(input.toString())
+
+                if (schemaType == SchemaType.FHIR_FHIR) {
+                    val configSchema = ConfigSchemaReader.readOneYamlSchema(
+                        transform.toStream(),
+                        FhirTransformSchema::class.java
+                    )
+                    val fhirTransform = if (configSchema is FhirTransformSchema) {
```

Comment:
How are we able to detect what kind of schema we are dealing with. I can change it to use like Report.format or something but we will need to update it anyways if we add CSV because there will be a different converter to use